### PR TITLE
[SGTM Docs] Dynamodb docs

### PR DESCRIPTION
**Pull request by vn6**
### Summary

- add documentation on dynamodb tables that SGTM uses

Asana tasks:
https://app.asana.com/0/1149418478823393/1207031348655582/f

Relevant deployment: 

CC: @suzyng83209 @harshita-gupta @prebeta @michael-huang87

### Risks

Pull Request: https://github.com/Asana/SGTM/pull/167

Pull Request synchronized with [Asana task](https://app.asana.com/0/0/1207068965339607)

### CODE REVIEW

**Review by harshita-gupta — COMMENTED** [review]
cool!

File: docs/dynamodb.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Dynamodb in SGTM
+
+This is an overview of how SGTM interacts with Dynamodb.
+
+## Tables
+
+#### sgtm-lock
+A table containing all locks for current SGTM processes. Locks are a GitHub id, currently either PR id or issue comment id.
```
Comment (suzyng83209) [inline docs/dynamodb.md:8]:
```suggestion
A table containing all locks for current SGTM processes. Locks are a GitHub node id, currently either PR id or issue comment id.
```

File: docs/dynamodb.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+To lock operations, use the `dynamodb_lock` helper:
+```
+from src.dynamodb.lock import dynamodb_lock
+
+with dynamodb_lock(pull_request_id):
+    ...
+```
+
+#### sgtm-objects
+A mapping of GitHub objects (github node string) to Asana objects (asana id string).
```
Comment (suzyng83209) [inline docs/dynamodb.md:21]:
```suggestion
A mapping of GitHub objects (github node id) to Asana objects (asana gid).
```

**Review by suzyng83209 — APPROVED** [review]
minor nits. Also should we mention that the `sgtm-users` table will be deprecated? Or will @harshita-gupta be making a doc change in this alongside that change?

**Comment by harshita-gupta** [comment]
I'm happy to add the docs edit into #166 once #167 merges!

**Pull request merged**